File: day-1/solution.py
class Solution:
	def threeStepsClosest(self, s: List[int], X: int) -> int:
    	s.sort()
    	b = sum(s[:3])
    	n = len(s)
    	for i in range(n):
        	l = i + 1
        	r = n - 1
        	while l < r:
            	s1 = s[i] + s[l] + s[r]
            	if s1 == X:
                	return s1
            	if abs(s1 - X) < abs(b - X):
                	b = s1
            	if s1 < X:
                	l += 1
            	else:
                	r -= 1
    	return b

File: day-1/solution.java
class Solution {
	public int threeStepsClosest(int[] steps, int X) {
    	Arrays.sort(steps);
    	int returnSum = Integer.MAX_VALUE;
    	int minDiff = Integer.MAX_VALUE;
    	for (int i = 0; i < steps.length - 2; i++) {
   		 int j = i+1;
   		 int k = steps.length - 1;
   		 while(j<k) {
   			 int sum = steps[i] + steps[j] + steps[k];
   			 int diff = Math.abs(X - sum);
   			 if(sum==X) {
   				 return sum;
   			 }
   			 returnSum = minDiff>diff?sum:returnSum;
            	minDiff = Math.min(minDiff, diff);
   			 if(sum<X) {
   				 j++;
   			 } else{
   				 k--;
   			 }
   		 }
   	 }
    	return returnSum;
	}
}

